core/src/main/java/org/apache/carbondata/core/scan/result/impl/FilterQueryScannedResult.java

```diff
@@ -54,7 +55,28 @@ public FilterQueryScannedResult(BlockExecutionInfo tableBlockExecutionInfos,
   }
 
   @Override public List<byte[]> getDictionaryKeyArrayBatch(int batchSize) {
-    throw new UnsupportedOperationException("Operation not supported");
+    // rowId from where computing need to start
```

Comment:
This method seems duplicated. Move this method to `BlockletScannedResult` and create an abstract method `fillValidRowIdsBatchFilling` and override in both implementation classes.

Reply:
Done.